junit5/src/main/java/cz/xtf/junit5/listeners/ProjectCreator.java

```diff
@@ -53,17 +93,71 @@ public void visit(NamespaceBuilder builder) {
         if (OpenShiftConfig.pullSecret() != null) {
             openShift.setupPullSecret(OpenShiftConfig.pullSecret());
         }
+        log.info("Created namespace: " + openShift.getNamespace());
+    }
+
+    @Override
+    public void afterAll(ExtensionContext context) {
+        if (OpenShiftConfig.useNamespacePerTestcase() && JUnitConfig.cleanOpenShift()) {
+            deleteProject(OpenShiftConfig.namespace(), false);
+        }
     }
 
     @Override
     public void testPlanExecutionFinished(TestPlan testPlan) {
         if (JUnitConfig.cleanOpenShift()) {
-            boolean deleted = openShift.deleteProject();
-            // For multi-module maven projects, other modules may attempt to crate project requests immediately after this modules deleteProject
-            if (deleted) {
-                BooleanSupplier bs = () -> openShift.getProject() == null;
-                new SimpleWaiter(bs, TimeUnit.MINUTES, 2, "Waiting for old project deletion").waitFor();
+            deleteProject(XTFConfig.get(OPENSHIFT_NAMESPACE), true);
+        }
+    }
+
+    private void deleteProject(String namespace, boolean waitForDeletion) {
+        OpenShift openShift = OpenShifts.master(namespace);
+        boolean deleted = openShift.deleteProject();
+        log.info("Start deleting namespace: " + openShift.getNamespace());
+        if (!deleted && waitForDeletion) {
+            BooleanSupplier bs = () -> openShift.getProject() == null;
+            new SimpleWaiter(bs, TimeUnit.MINUTES, 2, "Waiting for " + openShift.getNamespace() + " project deletion")
+                    .waitFor();
+        }
+    }
+
+    @Override
+    public FilterResult apply(TestDescriptor testDescriptor) {
+        if (testDescriptor instanceof MethodBasedTestDescriptor) {
+            boolean disabled = Arrays.stream(((MethodBasedTestDescriptor) testDescriptor).getTestClass().getAnnotations())
+                    .filter(annotation -> annotation instanceof Disabled).count() > 0;
+            if (!disabled) {
+                OpenShiftConfig.getTestcaseToNamespaceMap().putIfAbsent(
```

Comment:
I have strong feeling this can be simplified to new OpenShiftConfig method with TestDescriptor parameter e.g. something like `OpenShiftConfig.matchTestSuiteToNamespace(testDescriptor)` (name can be improved)

That would mean to move `getNamespaceForTestClass` to `OpenShiftConfig`.

Reply:
Can it be set as resolved?

Reply:
Yes, my understanding is this was addressed.